### PR TITLE
CI: Add Pyright CI

### PR DESCRIPTION
**Pull request by webmiche**
Sasha: stealing this PR

This workflow introduces a Pyright run for each PR, that checks for errors introduced anywhere in the project. Unlike the other Pyright CI, it does not display them inline, so the other one is still useful.

I added a list of currently failing files, with 521 errors in total. The CI will not run on those. The aim would be to hunt them down later, and remove them from the list of ignored files one-by-one. Once the list is empty we can delete `pyright-ci.json`

The json file cannot be put in a directory, I tried that, and it messes with the python path...

### CODE REVIEW

**Comment by superlopuh** [comment]
Nice! Looks like we have some pyright issues to get rid of still :)

File: .github/workflows/ci-pyright.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
```
Comment (georgebisbas) [inline .github/workflows/ci-pyright.yml:1]: can drop single version of Python

Comment (math-fehr) [reply .github/workflows/ci-pyright.yml:1]: Should we keep only `3.10`?
That way, we are sure we don't use features only available from `3.11`.

**Comment by superlopuh** [comment]
Oh sorry I forgot to rename my file

**Comment by webmiche** [comment]
np. Yours is probably a better approach to the CI anyway. I guess this PR is mostly for us to look at and understand how much is failing atm...

**Comment by webmiche** [comment]
Since the last run, we pretty much double our pyright errors. That's not great...

**Comment by superlopuh** [comment]
Quite a lot are in the new parser, I expect it wouldn't be that much work to get rid of them, but it's not something that we could've caught before the CI anyway :) We can't really expect all contributors to have the same coding style if it's not enforced by CI

**Comment by math-fehr** [comment]
I have somewhat of an idea to make this a smoother transition.
Could we try to add on the Pyright CI all files that are currently passing, to be sure that we are not going back? The current system (showing the CI failures on the PR) does not actually show it on the comments, meaning that we easily forget them.
Then, we add files one by one on this CI, until we pass everything.
What do you think?

**Comment by tobiasgrosser** [comment]
We did that in other projects and it worked pretty well.

**Comment by webmiche** [comment]
We already have the review dog thingi that comments on new pyright errors. I think this is to me more of a check of what the current count is then really something that we will merge in the close future. Not sure what others think...

**Comment by superlopuh** [comment]
!! Pyright CI passes! Stealing this PR :)

**Review by math-fehr — APPROVED** [review]
Yep, I think this is the way to go!

**Pull request merged**